jupyterhub/proxy.py

```diff
@@ -384,6 +399,11 @@ async def check_routes(self, user_dict, service_dict, routes=None):
                     )
                     futures.append(self.add_service(service))
 
+        # Add extra routes we've been configured for
+        for routespec, url in self.extra_routes.items():
+            good_routes.add(routespec)
+            futures.append(self.add_route(routespec, url, {'hub': True}))
```

Comment:
Not 'hub'? These aren't routes for the hub itself.

Maybe:

```suggestion
            futures.append(self.add_route(routespec, url, {'extra': True})
```

We mostly don't use this info for things other than user servers.

Reply:
Done

Reply:
@yuvipanda, the extra_routes will be auto-deleted after a while, what's the reason?